R/tree.R

```diff
@@ -31,200 +31,284 @@
-    prot_aln$seq
-    prot_dist <- dist.alignment(prot_aln, matrix = "similarity")
-    prot_nj <- NJ(prot_dist)
-    prot_nj_tree <- plot(prot_nj, main = "Neighbor Joining")
-    write.tree(phy = prot_nj_tree, file = tre_path)
+    if (enable_fast) {
+        # FastTree approach
+        print(fa_file)
+        system2(
+            command = fasttree_path,
+            args = paste(c(fa_file, ">", out_file),
+                sep = "", collapse = " "
+            )
+        )
+        ################################
+        ## Compiling FastTree.c on a Mac
+        ################################
+        ## Other platforms? Check http://www.microbesonline.org/fasttree/
+        # system(paste("gcc -DNO_SSE -O3 -finline-functions -funroll-loops -Wall -o",
+        #              here("src/FastTree"),
+        #              here("src/FastTree.c"), "-lm", collapse=" "))
+    } else {
+        # Detailed phylogenetic analysis approach
+        ###########################
+        ## Approach 1
+        ###########################
+        ## https://a-little-book-of-r-for-bioinformatics.readthedocs.io/en/latest/src/chapter5.html
+        prot_aln <- read.alignment(file = fa_file, format = format)
+        prot_aln$seq
+        prot_dist <- dist.alignment(prot_aln, matrix = dist_matrix)
+        prot_nj <- NJ(prot_dist)
+        prot_nj_tree <- plot(prot_nj, main = "Neighbor Joining")
+        write.tree(phy = prot_nj_tree, file = out_file)
 
-    ###########################
-    ## Approach 2
-    ###########################
-    ## Alignment file formats and conversion
-    # read in sequence data, convert to phyDat
-    prot_fa <- read.phyDat(fa_file, format = "fasta", type = "AA")
-    prot_phyDat <- phyDat(prot_fa, type = "AA", levels = NULL)
-    prot10 <- subset(prot_phyDat, 1:10)
-    prot10_phyDat <- phyDat(prot10, type = "AA", levels = NULL)
+        ###########################
+        ## Approach 2
+        ###########################
+        ## Alignment file formats and conversion
+        # read in sequence data, convert to phyDat
+        prot_fa <- read.phyDat(fa_file, format = format, type = seq_type)
+        prot_phyDat <- phyDat(prot_fa, type = seq_type, levels = NULL)
+        prot_subset <- subset(prot_phyDat, 1:subset_size)
+        prot_subset_phyDat <- phyDat(prot_subset, type = seq_type, levels = NULL)
 
-    ## Model Testing & Distance Matrices
-    ## Comparison of different nucleotide or amino acid substitution models
-    mt <- modelTest(prot10, model = "all")
-    print(mt)
+        ## Model Testing & Distance Matrices
+        ## Comparison of different nucleotide or amino acid substitution models
+        mt <- modelTest(prot_subset, model = model_test)
+        print(mt)
 
-    # estimate a distance matrix using a Jules-Cantor Model
-    dna_dist <- dist.ml(prot10, model = "JC69")
+        # estimate a distance matrix using a Jules-Cantor Model
+        dna_dist <- dist.ml(prot_subset, model = dna_model)
```

Comment:
```suggestion
        prot_dist <- dist.ml(prot_subset, model = mt)
```

I know the original outdated code references DNA here, but I believe this should all be protein based on the inputs. If the change to prot_dist is correct, the code below will also need modification to reflect the variable name changes.

Reply:
@jananiravi : Can you confirm if the above can be considered correct?

Reply:
@epbrenner are these the unused tree generators? Using dna could be a stub? Maybe change to protein for now (will work with current molevolvr) or make it a function argument to take dna or protein fasta sequences -- future-proofing the function given starting with DNA seqs will be part of molevolvr sooner rather than later. Does that answer your question, @Klangina ?

Reply:
Sure @jananiravi, I will create an argument to switch between DNA and protein sequences and keep the default as protein. Will update the pr shortly.